src/pages/signin/index.tsx

```diff
@@ -18,33 +22,90 @@ const Layout = styled.div`
   flex-direction: column;
   align-items: center;
   margin: 0 auto;
+  gap: 30px;
+
+  @media (max-width: 767px) {
+    width: 325px;
+  }
 `;
 
 const Form = styled.form`
+  width: 100%;
   display: flex;
   flex-direction: column;
   gap: 30px;
 `;
+
 function SigninPage() {
+  const router = useRouter();
+
+  const {
+    register,
+    formState: { errors },
+    handleSubmit,
+    setError,
+  } = useForm({ mode: "onBlur" });
+
+  const emailValidation = register("userEmail", {
+    required: {
+      value: true,
+      message: "이메일을 입력해 주세요",
+    },
+    pattern: {
+      value: /^\S+@\S+$/i,
+      message: "올바른 이메일 주소가 아닙니다.",
+    },
+  });
+
+  const passwordValidation = register("userPw", {
+    required: {
+      value: true,
+      message: "비밀번호를 입력해 주세요",
+    },
+  });
+
+  const onSubmit = async (data: any) => {
+    const res = await loginRequest(data);
+
+    if (res) {
+      const { data } = res;
+      window.localStorage.setItem("accessToken", data.accessToken);
+      router.push("/folder");
+    } else {
+      setError("userEmail", { message: "이메일을 확인해주세요" });
+      setError("userPw", { message: "비밀번호를 확인해주세요" });
+    }
+  };
+
   return (
     <BackGround>
       <Layout>
-        <LogoBox />
-        <Form action="">
-          <Input
-            label="이메일"
-            id="user_email"
-            type="email"
-            placeholder="이메일을 입력해 주세요"
-          ></Input>
-          <Input
-            label="비밀번호"
-            id="user_pw"
-            type="password"
-            placeholder="비밀번호를 입력해 주세요"
-          ></Input>
+        <LogoBox text="회원이 아니신가요?" linkText="회원 가입하기" />
+        <Form onSubmit={handleSubmit(onSubmit)}>
```

Comment:
r: 폼 요소의 에러 처리는 `handleSubmit`의 두번째 인자인 `onError`로 처리하면 함수의 책임이 분리되어 더 유연한 코드가 될 것 같아요.
```suggestion
        <Form onSubmit={handleSubmit(onSubmit, onError)}>
        {/* 잘못된 값으로 폼 제출 시 `register` 함수에서 정의한 에러 메시지를 `onError`에서 전달받아 사용하기 /*}
```